File: backend/app/api/schemas/me.py
"""Схемы для self-service эндпоинтов (T-424)."""

from __future__ import annotations

from pydantic import BaseModel


class ModelUsageBreakdown(BaseModel):
    """Расход по конкретной модели за текущий месяц."""

    model_id: str
    requests: int
    tokens_in: int
    tokens_out: int
    cost: float


class MyUsageResponse(BaseModel):
    """Личный расход пользователя в текущем месяце (T-424, T-435).

    tokens_limit / cost_limit: None = unlimited по этому измерению.
    Frontend проверяет limit === null per-field для отображения "Без лимита".
    near_limit: True когда использовано >= budget_near_limit_threshold (T-435).
    При budget=None (unlimited) — всегда False.
    """

    tokens_used: int
    tokens_limit: int | None
    cost_used: float
    cost_limit: int | None
    period: str
    by_model: list[ModelUsageBreakdown]
    near_limit: bool = False

File: frontend/src/api/me.ts
import { apiFetch } from "./client";
import type { ApiError } from "./runtime";

export interface ModelUsageBreakdown {
  model_id: string;
  requests: number;
  tokens_in: number;
  tokens_out: number;
  cost: number;
}

export interface MyUsageResponse {
  tokens_used: number;
  tokens_limit: number | null;
  cost_used: number;
  cost_limit: number | null;
  period: string;
  by_model: ModelUsageBreakdown[];
  near_limit: boolean;
}

export async function apiGetMyUsage(): Promise<MyUsageResponse> {
  return apiFetch<MyUsageResponse>("/api/auth/me/usage");
}

export type { ApiError };
